nachomemes/bot.py

```diff
@@ -107,124 +192,99 @@ async def on_message_edit(before: Message, after: Message):
-    return "**Potential Templates**" + "".join(lines)
-
-
-
-
-def list_templates(config: GuildConfig) -> str:
-    memes = STORE.list_memes(config.guild_id, ("name", "description"))
-    lines = [f"\n{m['name']}: *{m['description']}*" for m in memes]
-    return "**Templates**" + "".join(lines)
-
-
 @bot.group(description="Administrative functions group")
 async def memebot(ctx: Context):
     if ctx.subcommand_passed:
         return
     try:
-        message = "\n".join((f"**{s.name}**: {s.description}"
+        message = "\n".join((f"**`{s.name}`**: {s.description}"
             for s in ctx.command.walk_commands()))
-        await ctx.send("Available commands\n"+message)
+        return await ctx.send(embed=Embed(
+            title = "Available commands", 
+            description = message
+        ))
     except Exception as ex:
         await report(ctx, ex, "error listing commands")
 
 
 @memebot.command(description="updates the database with template data")
 async def refresh(ctx: Context, refresh_type: str=None):
+    is_hard = refresh_type == "--hard"
+    title = "Executinfg hard refresh" if is_hard else "Executing refresh"
     try:
         await ctx.trigger_typing()
-        is_hard = refresh_type == "--hard"
         config: GuildConfig = STORE.guild_config(ctx.guild)
         if is_hard:
             if not config.can_admin(_get_member(ctx)):
-                return await ctx.send(
-                    f"```{config.no_admin(_get_member(ctx))}```"
-                )
+                return await ctx.send(embed=Embed(
+                    title = title, 
+                    description = config.no_admin(_get_member(ctx))
+                ))
         else:
             if not config.can_edit(_get_member(ctx)):
-                return await ctx.send(
-                    f"```{config.no_admin(_get_member(ctx),'refresh','edit')}```"
-                )
-
-        message = STORE.refresh_memes(config.guild_id, is_hard)
-        await ctx.send(f"```{message}```")
+                return await ctx.send(embed=Embed(
+                    title = title, 
+                    description = config.no_admin(_get_member(ctx),'refresh','edit')
+                ))
+        return await ctx.send(embed=Embed(
+                title = title, 
+                description = STORE.refresh_memes(config.guild_id, is_hard)
+        ))
     except Exception as ex:
         await report(ctx, ex)
 
+
 @memebot.command(description="set the discord role for adminstrators")
 async def admin_role(ctx: Context, role_id: str=None):
     try:
         config: GuildConfig = STORE.guild_config(ctx.guild)
         if not role_id:
             role: Optional[Role] = ctx.guild.get_role(config.admin_role) if config.admin_role else None
-            await ctx.send(textwrap.dedent(
-                f"```Members of '{role}' are authorized to administer the memes.```"))
+            return await ctx.send(embed=Embed(
```

Comment:
Can we make a more descriptive error message if no admin role is setup instead of just putting `None` in there?